DuckDuckGo/AppDelegate.swift

```diff
@@ -42,18 +50,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         testing = ProcessInfo().arguments.contains("testing")
         if testing {
             window?.rootViewController = UIStoryboard.init(name: "LaunchScreen", bundle: nil).instantiateInitialViewController()
+            return true
         }
-
+        
+        // only replace an existing window in case starting up for background mode
+        if window != nil {
+            let window = TouchWindow()
+            window.makeKey()
+            window.rootViewController = self.window?.rootViewController
+            self.window = window
+        }
+        
+        EasyTipView.updateGlobalPreferences()
         HTTPSUpgrade.shared.loadDataAsync()
         
         // assign it here, because "did become active" is already too late and "viewWillAppear"
         // has already been called on the HomeViewController so won't show the home row CTA
         AtbAndVariantCleanup.cleanup()
         DefaultVariantManager().assignVariantIfNeeded()
         HomePageConfiguration.installNewUserFavorites()
-        
-        if !testing {
-            autoClear = AutoClear(worker: mainViewController!)
+
+        if let main = mainViewController {
```

Comment:
Can't depend on mainViewController being set if app has started in the background.